wordpress-mobile/WordPress-iOS
# Fix a crash when multiple instances of Reader are opened after tapping on blogging prompts

Fixes #20843

Reader topics view can be opened multiple times since `ReaderCoordinator.showTag()` performs an async network call to fetch a topic before presenting the controller. If users have a slower network connection, they can easily click multiple times, triggering multiple network calls, multiple completion blocks, and duplicating navigation in Reader, which eventually causes a crash.

### Solution:

Use already existing `controllerWithTagSlug` that makes the same `service.tagTopicForTag` async network request to fetch the topic after presenting Reader. It makes `showTag()` synchronous and prevents viewController from opening multiple items.

Other possible solutions:
- Disable the blogging prompt button while an async call is being made
- Show loading indicator and disable UI interaction while an async call is being made

I decided that these alternative solutions are less responsive and provide worse user experience.

**Note:** since similar issues are also logged in other places in the app (such as Stats), I will look for similar interactions in other areas of the app.

## To test:

1. Log in to a WP.com account
2. On the dashboard, tap the blogging prompts "View all responses" multiple times in quick succession
3. Reader topic view should open and eventually load a daily prompt topic
4. Navigate back
5. App shouldn't crash

## Regression Notes
1. Potential unintended areas of impact

Daily prompt navigation and prompt showing not working

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual tests

3. What automated tests I added (or what prevented me from doing so)

None

PR submission checklist:


UI Changes testing checklist:
